src/main/java/com/minecolonies/coremod/colony/buildings/modules/AbstractCraftingBuildingModule.java

```diff
@@ -670,19 +670,36 @@ public boolean fullFillRecipe(final IRecipeStorage storage)
             return storage.fullfillRecipe(building.getColony().getWorld(), handlers);
         }
         final AbstractEntityCitizen worker = data.getEntity().get();
-        final int primarySkill =worker.getCitizenData().getCitizenSkillHandler().getLevel(building.getModuleMatching(WorkerBuildingModule.class, m -> m.getJobEntry() == jobEntry).getPrimarySkill());
-        final int luck = (int)(((primarySkill + 1) * 2) - Math.pow((primarySkill + 1 ) / 10.0, 2));
 
         LootContext.Builder builder =  (new LootContext.Builder((ServerLevel) building.getColony().getWorld())
                                           .withParameter(LootContextParams.ORIGIN, worker.position())
                                           .withParameter(LootContextParams.THIS_ENTITY, worker)
-                                          .withParameter(LootContextParams.TOOL, worker.getMainHandItem())
+                                          .withParameter(LootContextParams.TOOL, getCraftingTool(worker))
                                           .withRandom(worker.getRandom())
-                                          .withLuck((float) luck));
+                                          .withLuck(getCraftingLuck(worker)));
```

Comment:
Don't we have the fortune already, not sure what the luck is for then.

Reply:
We don't use luck in this case. Luck is for other loottable scenarios, where there isn't a tool. The future netherworker for example.